### ISSUE TITLE
Find correct location of declared packages

### ISSUE THREAD

**Issue by blackheaven** [blackheaven/haskell-security-action#5]
Currently, `Region`s are hard-coded, we can give the exact line of the issue.

Related to #1 

While it should be given by `cabal-audit`, we maybe be able to give an approximation ourselves.

**Comment by dancewithheart** [comment]
I think PR https://github.com/MangoIV/cabal-audit/pull/75 should address most of this. I marked it as fixing this issue.
Please correct me if I am overstating it or missing an important case :)

It is not perfect. I would like to do a follow up:
- consider lock/config files from Nix e.. `flake.lock`
- consider Stack files: `stack.yaml.lock`, `stack.yaml`
- maybe expose additional matches as `relatedLocations`

That could happen along side https://github.com/MangoIV/cabal-audit/issues/68.

From using `cabal-audit` via `haskell-security-action` lack of transitive dependency handling feels like the main pain point.

**Comment by blackheaven** [comment]
I agree, but that's already a big first step, I think it'll be improved over time.

Again, thanks for your hard job!
